### ISSUE TITLE
Explore easier sharing mechanisms

### ISSUE THREAD

**Issue by parlough** [dart-lang/dart-pad#2733]
It would be nice if whatever used here could also be used by "Open in DartPad" [in lightweight DartPads](https://github.com/dart-lang/dart-pad/issues/2702#issuecomment-1791322473).

A few ways:

**Upload snippet (up to max-length) to some database or blob storage**

- Easy and flexible, but has potential concerns due to storing user-generated content (although anonymous)
- Can automatically delete after not being accessed for X-days (entries can be exempted)
- Easier to share than using a query parameter
- The [Go Playground](https://go.dev/play/) uses [datastore](https://github.com/golang/playground/blob/master/store.go)

**Query parameter:**

- Has size limitations?
- No need to store anything on server

Extra:

- Besides using query parameters, the [Rust playground](https://play.rust-lang.org/) seems to also support uploading to a shared [GitHub gist account](https://gist.github.com/rust-play)?

**Comment by parlough** [comment]
As a note, the Flutter API docs could then migrate to use the shared mechanism rather than the custom solution we have for them now.

**Comment by alisheraxmedov** [comment]
Sezar

**Comment by OnkelTem** [comment]
Hi folks.

I'm new to DartPad and I was really surprised that I basically cannot share my code. 
Sure, I read the Sharing Guide but it doesn't seem to be usable - making gists, copy long numbers, omg...

Can I share my code anyway like via a query parameter?

**Comment by OnkelTem** [comment]
> * Hard to actually share. Potentially too large for sharing in some chat areas and social media platforms

Of course. Yet, we can always use some trusted URL shorteners: let the size limit be their database problems :)

**Comment by Levi-Lesches** [comment]
@parlough Would you happen to know why the automatic Gist creation feature is no longer present? I thought it was pretty neat, being able to sign in with GitHub and pressing a button to save and share your code.
